crates/ark/src/variables/variable.rs

```diff
@@ -1336,6 +1433,88 @@ impl PositronVariable {
 
         Ok(out)
     }
+
+    fn try_inspect_custom_method(value: SEXP) -> Result<Option<Vec<Variable>>, harp::Error> {
+        let result: Option<RObject> = ArkGenerics::VariableGetChildren
+            .try_dispatch(value, vec![])
+            .map_err(|err| harp::Error::Anyhow(err))?;
+
+        match result {
+            None => Ok(None),
+            Some(value) => {
+                // Make sure value is a list before using inspect_list
+                if !r_typeof(value.sexp) == LISTSXP {
+                    return Err(harp::Error::Anyhow(anyhow!(
+                        "Expected `{}` to return a list.",
+                        ArkGenerics::VariableGetChildren.to_string()
+                    )));
+                }
+
+                // This is essentially the same as Self::inspect_list but with modified `access_key`
+                // that adds more information about the object:
+                // 1. Provide the name and the index for the `get_child_at` method.
+                // 2. (Not necessary) Given an access key, we can detect if we want to apply a custom get_child_method.
+                let list = List::new(value.sexp)?;
+                let n = unsafe { list.len() };
+
+                let names = local!({
+                    let r_names = unsafe { RObject::new(Rf_getAttrib(value.sexp, R_NamesSymbol)) };
+                    if r_is_null(r_names.sexp) {
+                        return vec![None; n];
+                    }
+
+                    let names = unsafe { CharacterVector::new_unchecked(r_names) };
```

Comment:
Simplify with `RObject::attr()` or `RObject::names()`?

Reply:
Done in https://github.com/posit-dev/ark/pull/561/commits/c7ae20e423fd3921d88fef90c12926dfc156cf4a